### PR TITLE
Convert update to interactions

### PR DESCRIPTION
**Pull request by alexshtin**
**What changed?**
Convert Workflow update functionality to interactions.

**Why?**
New design proposal.

**Breaking changes**
Yes, but update is not released yet.

### CODE REVIEW

File: temporal/api/enums/v1/interaction_type.proto [hunk 1]
```diff
@@ -22,25 +22,18 @@
...
-    // Headers that are passed with the update to the processing workflow.
-    // These can include things like auth or tracing tokens.
-    temporal.api.common.v1.Header header = 1;
-
-    // The name of the update function to invoke on the target workflow.
-    string name = 2;
-
-    // The arguments to pass to the named update function.
-    temporal.api.common.v1.Payloads args = 3;
+enum InteractionType {
```
Comment (bergundy) [inline temporal/api/enums/v1/interaction_type.proto:34]: I thought we're calling these messages now.. did that change?

**Review by bergundy — COMMENTED** [review]
This doesn't match what we discussed yesterday in our design session.

File: temporal/api/enums/v1/interaction_type.proto (hunk 1 above)
Comment (mmcshane) [reply temporal/api/enums/v1/interaction_type.proto:34]: This is an intermediate step. Alex has some code that we want to get working end-to-end but it hasn't adopted the protocol/message paradigm yet. We want to be refactoring from a known-working base state.

**Review by mmcshane — APPROVED** [review]

**Comment by alexshtin** [comment]
@bergundy, this is v2 of our design. What we discussed is v3. We might postpone v3 to after upcomming server release (1.19).

**Review by yiminc — APPROVED** [review]

**Comment by bergundy** [comment]
Got it, thanks for the update, this came out of the blue for me.

File: temporal/api/workflowservice/v1/request_response.proto [hunk 2]
```diff
@@ -304,6 +306,8 @@ message RespondWorkflowTaskCompletedResponse {
     PollWorkflowTaskQueueResponse workflow_task = 1;
     // See `ScheduleActivityTaskCommandAttributes::request_start`
     repeated PollActivityTaskQueueResponse activity_tasks = 2;
+
+    int64 reset_history_event_id = 3;
```
Comment (Sushisource) [inline temporal/api/workflowservice/v1/request_response.proto:310]: This already got merged, but, this really deserves a docstring.

**Pull request merged**